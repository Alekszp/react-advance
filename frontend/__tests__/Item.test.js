import ItemComponent from '../components/Item'
import {shallow} from 'enzyme'
import toJSON from 'enzyme-to-json'

const fakeItem = {
    id: 'abcd123',
    title: 'some title',
    price: 2121,
    description: "very good descriptionnn",
    image: 'dog.jpeg',
    largeImage: 'big_dog.jpeg'
}

describe('<Item />', ()=>{
    it('renders and matches the snapshot',()=>{
        const wrapper = shallow(<ItemComponent item={fakeItem} />)
        expect(toJSON(wrapper)).toMatchSnapshot()
    })


    // it('renders and displays properly', ()=>{
    //     const wrapper = shallow(<ItemComponent item={ fakeItem } />)
    //     const PriceTag = wrapper.find('PriceTag')
    //     // console.log(PriceTag.debug())
    //     // console.log(PriceTag.text())
    //     // console.log(PriceTag.dive().text() )
    //     // console.log(PriceTag.children().text())
    //     expect(PriceTag.children().text()).toBe('$21.21')
    //     expect(wrapper.find('Title a').text()).toBe(fakeItem.title)
        

    // })
    // it('renders the image properly', ()=>{
    //     const wrapper = shallow(<ItemComponent item={ fakeItem } />)
    //     const img = wrapper.find('img')
    //     expect(img.props().src).toBe(fakeItem.image)
    //     expect(img.props().alt).toBe(fakeItem.title)

    // })
    // it('renders out the buttons properly', ()=>{
    //     const wrapper = shallow(<ItemComponent item={fakeItem} />)
        
    //     const buttonList = wrapper.find('.buttonList')
    //     expect(buttonList.children()).toHaveLength(3)
    //     expect(buttonList.find('Link')).toHaveLength(1) 
    //     //OR
    //     expect(buttonList.find('Link').exists()).toBe(true)
    //     //OR
    //     expect(buttonList.find('Link')).toBeTruthy()
    //     expect(buttonList.find('AddToCart')).toBeTruthy()
    //     expect(buttonList.find('DeleteFromCart')).toBeTruthy()
    // })
})