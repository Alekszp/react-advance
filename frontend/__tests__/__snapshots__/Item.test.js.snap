// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="some title"
    src="dog.jpeg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "abcd123",
          },
        }
      }
    >
      <a>
        some title
      </a>
    </Link>
  </Title>
  <PriceTag>
    $21.21
  </PriceTag>
  <p>
    very good descriptionnn
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "abcd123",
          },
        }
      }
    >
      <a>
        Edit 
        <Item__Icon>
          ‚úè
        </Item__Icon>
      </a>
    </Link>
    <AddToCart
      id="abcd123"
    />
    <DeleteItem
      id="abcd123"
    >
      Delete this item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
